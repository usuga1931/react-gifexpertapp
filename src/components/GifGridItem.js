import React from 'react';

export const GifGrifItem = ({ title, url }) => {

    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    );
}