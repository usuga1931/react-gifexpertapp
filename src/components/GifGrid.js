import React from 'react';
import { useFetchGif } from '../hooks/useFetchGifs';
import { GifGrifItem } from '../components/GifGridItem';

export const GifGrid = ({ category }) => {

    const { data:images, loading } = useFetchGif(category);

    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {loading && <p>Cargando</p>}
            <div className="card-grid animate__animated animate__flash">
                {
                    images.map(img => (
                        <GifGrifItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}
