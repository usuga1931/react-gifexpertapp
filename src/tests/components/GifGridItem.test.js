import React from 'react';
import { shallow } from 'enzyme';
import { GifGrifItem } from '../../components/GifGridItem';

describe('Pruebas con el componente GifGridItem', () => {

    const title = 'One Punch';
    const url = 'https://media1.giphy.com/media/6IkjQmpaRwIabJ2G3C/giphy-downsized-medium.gif?cid=8379dea4yjtir9dbixckxxqvsb2k14dm2t0ucsfb39oi87q7&rid=giphy-downsized-medium.gif';
    const wrapper = shallow(<GifGrifItem title={title} url={url} />);

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un pÃ¡rrafo con el title', () => {

        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);

    });

    test('Debe de tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    });

    test('Debe de tener animate__fadeIn', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);

    });
});
