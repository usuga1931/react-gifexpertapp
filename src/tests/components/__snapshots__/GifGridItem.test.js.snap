// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el componente GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="One Punch"
    src="https://media1.giphy.com/media/6IkjQmpaRwIabJ2G3C/giphy-downsized-medium.gif?cid=8379dea4yjtir9dbixckxxqvsb2k14dm2t0ucsfb39oi87q7&rid=giphy-downsized-medium.gif"
  />
  <p>
    One Punch
  </p>
</div>
`;
